pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/minion/PinotHelixTaskResourceManager.java

```diff
@@ -317,8 +317,10 @@ public synchronized String submitTask(String parentTaskName, List<PinotTaskConfi
   public synchronized Set<String> getTasks(String taskType) {
     String helixJobQueueName = getHelixJobQueueName(taskType);
     WorkflowConfig workflowConfig = _taskDriver.getWorkflowConfig(helixJobQueueName);
-    Preconditions.checkArgument(workflowConfig != null, "Task queue: %s for task type: %s does not exist",
-        helixJobQueueName, taskType);
+    if (workflowConfig == null) {
+      LOGGER.error("Task queue: {} for task type: {} does not exist", helixJobQueueName, taskType);
```

Comment:
Minor question: should this be an `ERROR` level log?  Would this condition indicate that the user gave bad information (an invalid task type) or that Helix is in a bad state (would it be a bad state if a task does not have an associated queue)?

I think if the issue is that the user gave an invalid value for `taskType` then the log level should be `INFO`. If the `taskType` is valid but the queue does not exist then that could be an `ERROR`.

Reply:
Good point, changed the log to info. I didn't find a way to check if the task is valid but not the queue. In fact, there are other places in the code which use task queue itself, for instance:
https://github.com/apache/pinot/blob/fb9f2953a2626c99412411a968f095bd0376521a/pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotTaskRestletResource.java#L185

Reply:
Yeah, I don't think it's possible to check if a task is valid, I'm pretty sure they can be dynamically defined (e.g. when Minion creates tasks). So I task type could be valid and not yet have a queue. But it could also never have a queue because no part of the system creates a task with the given type.